server/functions/basebuilding/fn_resupplybuilding.sqf

```diff
@@ -95,6 +95,23 @@ if (_resupplyType == "MULTISANDBAG" && _supplyType == "BuildingSupplies") then {
 //No supplies, no point continuing execution.
 if (_supplies == 0) exitWith {};
 
+private _currentSupplies = _supplySource getVariable ["para_g_current_supplies", 0];
+private _supplyConsumptionRate = _supplySource getVariable ["para_g_supply_consumption_rate", 1];
+
+private _proposedSuppliesSum = _currentSupplies + _supplies;
+private _proposedRemainingTimeSeconds = _proposedSuppliesSum / _supplyConsumptionRate;
+private _maxTimeLimit = 7200; // max time limit a building can have, 2 hours in seconds
+
+// Check if the current lifetime would exceed the time limit
+if ( _proposedRemainingTimeSeconds >= _maxTimeLimit) then {
+	// if the supplies that are being added will exceed 2 hours for the building, trim the supplies so that it only reaches 2 hours max. This will still consume the item!!!
+	_supplies = (_maxTimeLimit * _supplyConsumptionRate) - _currentSupplies;
+
+	// notify players that time limit has been reached and item was still consumed
+	["TaskFailed",["",localize "STR_para_build_maxtimereached"]] call para_c_fnc_show_notification;
```

Comment:
This will probably need remoteExec-ing for the specific player, which means we need to pass the player object into the args for this script. 

On player-hosted local development this will work because the server IS the player. But probably won't work on dedicated server. 

I think. On phone so bit limited in what I can check easily. 

Also, there's a notifications config here https://github.com/Bro-Nation/Paradigm/blob/development/client/configs/notifications.hpp

But that's for "client" functions and this is a "server" function. 

Hmmm. I Need to think about this/have a look at the rest of the code path. I wonder if we can do the "too long, are your supplies" notification on whatever script calls this on the client?

Because the interaction overlay already shows how much time is left for supplies, so that value is already available to clients.

Reply:
I was copying how they did notifications for adding a crate.

https://github.com/Bro-Nation/Paradigm/blob/development/client/functions/basebuilding/fn_resupply_building_with_crate.sqf

`if !(_nearbySupplies isEqualTo []) then
{
	["resupplybuilding", [_target, "CRATE"]] call para_c_fnc_call_on_server;
} else {
	["TaskFailed",["",localize "STR_vn_mf_nosupplydropnearby"]] call para_c_fnc_show_notification;
};
`

Last line, they call a notification in a very similar way so, I think, it ought work, I do see what you mean though.

Reply:
> This will probably need remoteExec-ing for the specific player, which means we need to pass the player object into the args for this script.

Classic viewing a PR on my phone, I missed the fact `_player` is already in here as a variable. See **REhandler note** below for more info.

What you'll need for this specific line (111) is

```sqf
[["TaskFailed",["",localize "STR_para_build_maxtimereached"]]] remoteExecCall ["para_c_fnc_show_notification", _player];
```

It probably won't work on player hosted (local development), but it will work on dedicated server.

I'll need to add this in paradigm (it should really be here in `paradigm/server/remoteExec` anyway but I was being lazy :/ ): 

https://github.com/Bro-Nation/Mike-Force/blob/development/mission/functions/core/helpers/fn_rExecServerToGlobal_playerHost_or_dedicated.sqf

> Server to global client remoteExec depending on whether execution is happening on a player hosted server or a dedicated server.

----

### REhandler note

At the top:

```sqf
//THIS IS A REHANDLER FUNCTION
//It can only be called from rehandler, as it relies on having _player defined.
```

https://github.com/Bro-Nation/Paradigm/blob/development/server/functions/rehandler/fn_rehandler.sqf

`_player` is taken from the `para_s_fnc_rehandler` function's scope (layman's version -- `para_s_fnc_rehandler` calls this script and the `_player` variable still exists so this function can use it, one of those weird sqf things).

Reply:
> I was copying how they did notifications for adding a crate.
>
> Last line, they call a notification in a very similar way so, I think, it ought work, I do see what you mean though.

Tally, seems like it's just one of those paradigm things. That's all fine then.

Reply:
> Tally, seems like it's just one of those paradigm things. That's all fine then.

So it does or does not need to change from:

`["TaskFailed",["",localize "STR_para_build_maxtimereached"]] call para_c_fnc_show_notification;`

to:

`[["TaskFailed",["",localize "STR_para_build_maxtimereached"]]] remoteExecCall ["para_c_fnc_show_notification", _player];`

??

Reply:
sorry, to clarify

> Tally, seems like it's just one of those paradigm things. That's all fine then.

this was specifically about `localize "STR_para_build_maxtimereached"`. just saying i didn't realise they used that method. Mike Force has the notification class name for the most part. [Example here](https://github.com/Bro-Nation/Mike-Force/blob/development/mission/functions/systems/attachments/fn_attachments_client_battery_monitor_job.sqf#L28).

> So it does need to change

yep, will need the `remoteExec` change.

Reply:
I think I see what you mean, but how does the notification on the resupply_with_crate/sandbag working then? 

`if !(_nearbySupplies isEqualTo []) then { ["resupplybuilding", [_target, "CRATE"]] call para_c_fnc_call_on_server; } else { ["TaskFailed",["",localize "STR_vn_mf_nosupplydropnearby"]] call para_c_fnc_show_notification; }; `